packages/scripts/utils/php-file-paths-plugin.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+const { validate } = require( 'schema-utils' );
+
+/**
+ * Internal dependencies
+ */
+const { getPhpFilePaths } = require( '../utils' );
+
+const phpFilePathsPluginSchema = {
+	type: 'object',
+	properties: {
+		props: {
+			type: 'array',
+			items: {
+				type: 'string',
+			},
+		},
+	},
+};
+
+/**
+ * The plugin recomputes PHP file paths once on each compilation. It is necessary to avoid repeating processing
+ * when filtering every discovered PHP file in the source folder. This is the most performant way to ensure that
+ * changes in `block.json` files are picked up in watch mode.
+ */
+class PhpFilePathsPlugin {
+	/**
+	 * PHP file paths from `render` and `variations` props found in `block.json` files.
+	 *
+	 * @type {string[]}
+	 */
+	static paths;
```

Comment:
Probably need to change this so there won't be any collisions between class instances.

Reply:
Maybe I'm being overly cautious here. My concern is basically that if these paths are stored as a `static` class variable, they might get polluted. However, I'm not sure if the latter is realistic. While I can think of `npm run dev` running in parallel with another job that builds some 3rd-party package, that might not be enough to pollute class-level vars.

(OTOH, making it non-static -- in the context of a Webpack plugin -- seems to be non-trivial.)

Reply:
It worked exactly the same before and no issues were reported. As far as I understand it, the webpack config is executed only once when you run `wp-scripts start`. It's when the plugin get instatiated, all subsequent changes to paths happen through th webpack lifecycle events.

Reply:
> It worked exactly the same before and no issues were reported. 

It wasn't used for Core blocks -- i.e. Gutenberg code itself -- before, so my concern was vaguely that e.g. running the GB build in watch mode might collide with running a wp-scripts command to build a third-party block.

> As far as I understand it, the webpack config is executed only once when you run `wp-scripts start`. It's when the plugin get instatiated, all subsequent changes to paths happen through th webpack lifecycle events.

Yeah, I guess we should be fine 🤔